attributes/default.rb

```diff
@@ -13,3 +13,8 @@
 
 default[:mysql_component][:sql_url] = nil
 default[:mysql_component][:hosts] = ["localhost"]
+
+
+local_ipv4 = node[:cloud][:local_ipv4].kind_of?(Array) ? node[:cloud][:local_ipv4].first : node[:cloud][:local_ipv4]
+default[:mysql][:bind_address] = node.attribute?(:cloud) && local_ipv4 ? local_ipv4 : node[:ipaddress]
```

Comment:
This could be rewritten as (better use `normal` override, because `default` is already defined in `mysql` cookbook):
```
normal[:mysql][:bind_address] = node.attribute?(:cloud) && Array(node[:cloud][:local_ipv4])[0] ? Array(node[:cloud][:local_ipv4])[0] : node[:ipaddress]
```

With such notation you do not need first line with `local_ipv4` format detection.